tests/testlib/s2n_kem_tests.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/s2n_kem.h"
+#include "tests/unit/s2n_nist_kats.h"
+#include "utils/s2n_mem.h"
+#include "utils/s2n_random.h"
+#include "utils/s2n_safety.h"
+
+#define SEED_LENGTH 48
+uint8_t kat_entropy_buff[SEED_LENGTH] = {0};
+struct s2n_blob kat_entropy_blob = {.size = SEED_LENGTH, .data = kat_entropy_buff};
+
+int kat_entropy(struct s2n_blob *blob)
+{
+    eq_check(blob->size, kat_entropy_blob.size);
+    blob->data = kat_entropy_blob.data;
+    return 0;
+}
+
+int s2n_test_kem_with_kat(const struct s2n_kem *kem, const char *kat_file_name)
+{
+    notnull_check(kem);
+
+    FILE *kat_file = fopen(kat_file_name, "r");
+    notnull_check(kat_file);
+
+    /* Client side variables */
+    uint8_t *ct, *client_shared_secret, *pk, *sk, *server_shared_secret, *pk_answer, *sk_answer, *ct_answer, *ss_answer;
```

Comment:
This declaration contains all the variables. Would be better to move it above the comment for the section.